apps/web/src/pages/admin/users/index.tsx

```diff
@@ -0,0 +1,215 @@
+
+  const { page, setPage, handleTableHeaderClick, sortBy, sortDirection } =
+    usePagination<UserSortField>(1, UserSortField.CreatedAt)
+
+  const rolesArray = useMemo<Role[]>(
+    () =>
+      Object.values(FirebaseClaim).map((id) => ({
+        id,
+        label: t(`users.roles.${id}`),
+      })),
+    []
+  )
+
+  const qp = getUsersFilterQuery({
+    page,
+    sortBy,
+    sortDirection,
+    pageSize: USERS_PER_PAGE,
+    search: searchDebounced,
+  })
+  const { data, isValidating } = useAuthApi<UserAccounts>(
+    `${urls.api.v1.admin.getUsers}?${qp}`
+  )
+  const users = data?.users || []
+  const total = data?.total || 0
+
+  const renderClaims = useCallback(
+    ({ item }: RendererProps) =>
+      rolesArray.map((role) => (
+        <CheckRole
+          key={`${item.externalId}-${role.id}`}
+          user={item}
+          role={role}
+        />
+      )),
+    []
+  )
+
+  const submitSearch = useCallback(
+    debounce((value) => {
+      setSearchDebounced(value)
+    }, 200),
+    []
+  )
+
+  const handleSearchChange = useCallback((value) => {
+    setSearch(value)
+    submitSearch(value)
+  }, [])
+
+  const columns: ColumnDefinitionType<UserAccount>[] = [
+    {
+      key: 'username',
+      name: t('users.table.Username'),
+      sortable: true,
+    },
+    {
+      key: 'email',
+      name: t('users.table.Email'),
+      sortable: true,
+    },
+    {
+      key: 'createdAt',
+      name: t('users.table.Created'),
+      renderer: ({ value }) =>
+        value ? new Date(value).toLocaleDateString(lang) : null,
+      sortable: true,
+    },
+    {
+      key: 'claims',
+      name: t('users.table.Role'),
+      renderer: renderClaims,
+    },
+  ]
+
+  const searchBox = (
+    <div className={'m-5'}>
+      <TextInput
+        handleChange={handleSearchChange}
+        placeholder={'Search by email or username'}
+        value={search}
+      />
+    </div>
+  )
+
+  const footer = (
+    <>
+      <div>
+        <Pagination
+          currentPage={page}
+          total={total}
+          pageSize={USERS_PER_PAGE}
+          setPage={setPage}
+        />
+      </div>
+      {total > 0 && <div>{total} records found</div>}
+    </>
+  )
+
+  return (
+    <AdminLayout pageTitle={t('common:pageTitles.Users')}>
+      <Breadcrumbs
+        breadcrumbs={[
+          { label: 'Admin', href: urls.admin.index },
+          { label: 'Users' },
+        ]}
+      />
+      <Panel
+        fullWidth
+        title={t('common:pageTitles.Users')}
+        contentRight={
+          isValidating && <RefreshIcon className="w-5 h-5 animate-spin" />
+        }
+        footer={footer}
+      >
+        {searchBox}
+        <div className="overflow-x-auto">
+          <Table<UserAccount>
+            columns={columns}
+            data={users}
+            onHeaderClick={handleTableHeaderClick}
+            sortBy={sortBy}
+            sortDirection={sortDirection as any}
```

Comment:
I have a change in my PR to fix this to avoid the `as any` part. Basically changed the `sortDirection` property to be of `SortDirection` type.

https://github.com/deptagency/algomart/pull/340/files#diff-a6f934ce01ca170b72e82ccf5c173d7c6e18a1720a9227064d27ec8a4a304878